src/System.CommandLine.Suggest/dotnet-suggest-shim.bash

```diff
@@ -1,30 +1,30 @@
 # dotnet suggest shell complete script start
 _dotnet_bash_complete()
 {
-    local fullpath=`type -p ${COMP_WORDS[0]}`
-    local escaped_comp_line=$(echo "$COMP_LINE" | sed s/\"/'\\\"'/g)
-    local completions=`dotnet-suggest get --executable "${fullpath}" --position ${COMP_POINT} -- "${escaped_comp_line}"`
+    local fullpath=$(type -p "${COMP_WORDS[0]}")
+    local completions=$(dotnet-suggest get --executable "${fullpath}" --position "${COMP_POINT}" -- "${COMP_LINE}")
 
+    # filter suggestions by word to complete.
+    local word="${COMP_WORDS[COMP_CWORD]}"
     local IFS=$'\n'
-    local suggestions=($(compgen -W "$completions"))
+    local suggestions=($(compgen -W "$completions" -- "$word"))
 
-    if [ "${#suggestions[@]}" == "1" ]; then
-        local number="${suggestions[0]/%\ */}"
-        COMPREPLY=("$number")
-    else
-        for i in "${!suggestions[@]}"; do
-            suggestions[$i]="$(printf '%*s' "-$COLUMNS"  "${suggestions[$i]}")"
-        done
+    # format suggestions as shell input.
+    for i in "${!suggestions[@]}"; do
+        suggestions[i]="$(printf '%q' "${suggestions[$i]}")"
+    done
 
-        COMPREPLY=("${suggestions[@]}")
-    fi
+    COMPREPLY=("${suggestions[@]}")
 }
 
 _dotnet_bash_register_complete()
 {
-    local IFS=$'\n'
-    complete -F _dotnet_bash_complete `dotnet-suggest list`
+    if command -v dotnet-suggest &>/dev/null; then
```

Comment:
This is to avoid printing errors if the tool was uninstalled and the script is still sourced.